src/main/kotlin/com/doist/gradle/changelog/ChangelogCommitTask.kt

```diff
@@ -0,0 +1,56 @@
+package com.doist.gradle.changelog
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.VerificationTask
+import org.gradle.kotlin.dsl.property
+
+@Suppress("UnstableApiUsage")
+abstract class ChangelogCommitTask : DefaultTask(), VerificationTask {
+    init {
+        description = "Adds pending changelog entries into the changelog file."
+        group = "changelog"
+    }
+
+    @get:InputDirectory
+    val pendingChangelogDir: DirectoryProperty = project.objects.directoryProperty()
+
+    @get:InputFile
+    val changelogFile: RegularFileProperty = project.objects.fileProperty()
+
+    @get:Input
+    val commitConfig: Property<CommitConfig> = project.objects.property()
+
+    @TaskAction
+    fun commitChangelogEntries() {
+        val pendingChangelogDir = pendingChangelogDir.asPendingChangelogDir() ?: return
+        val changelogFile = changelogFile.get().asFile
+
+        if (!changelogFile.exists()) {
+            runCatching {
+                changelogFile.mkdirs()
+                changelogFile.createNewFile()
+            }.onFailure {
+                throw GradleException("Cannot create ${changelogFile.name} file.")
+            }
+        }
```

Comment:
Currently if there is no changelog file, the task fails before this code can run with error:

```> File '%path_to_project%/CHANGELOG.md' specified for property 'changelogFile' does not exist.```

Reply:
Yeah, I think it's just simpler if we remove this part 😅